determined-ai/determined
# docs: address onboarding gaps

## Description

This PR fixes the issues that are brought up by @hoang-determined-ai 👍 

1. Write porting guides for different frameworks and put it in Training: Implement Training APIs. 
2. Move porting guide from tutorials to Training: Implement Training APIs 
- Remove pining GPUs 
- Metric reducers should be advanced sections of porting guides (lightly) and references (detailed) 
- Porting guides should include additional best practices and links out to more information in the appropriate sections: 
- Using custom loss, metrics (custom functions) 
- Using custom callbacks 
- Data: Working with different data sources (can link out to “Prepare data”) 
- Validation
- Using Metrics Reducer (can link out) 
2. Tutorials: Remove MNIST tutorials for Keras and Pytorch 
3. Add Concepts page 
4. Categorize Features in the Features page 
5. Move Notebooks, Shells, Commands, TensorBoards, Templates out of Features as a standalone first-level page. And put them under the Additional Capabilities title section. 
6. Move experiment page to be a section of Training: Implement Training APIs. 
- Tell about experiments. 
- Link to experiment cycle. 
- Why you need to implement training APIs? What is the purpose of it? 
- How to submit an experiment. 
7. Add non-clickable Training title section in table of contents. 
8. Add non-clickable SysAdmin title section in table of contents. 
9. Prepare Data (page): Revised it to be opinionated about option A. Just reordering should be fine. 
10. Interact with Cluster: Tell users that Python APIs and REST APIs are only for automation. 
11. Recommend what to use for checkpoints in Ecosystem Integration page - REST API & Python APIs 
13. Enforce Quick Start Guide to be “Getting Started with Examples”. And move it to the landing page of documents. 
- Install Docker  “Follow these instructions to … set up Docker if you haven’t installed it already” 
- Include a note before the gray example box, after the Docker instructions: 
- “The commands below demonstrate how to start a Determined cluster locally, run an example experiment, and access the Determined Web UI” 
- Install Determined 
- Install Determined locally 
- Note: if you plan to use Jupyter notebooks, set up persistent storage by doing X 
- Advanced: install Determined on-premise, AWS, GCP (links) 
- Comment: “installation instructions” leads to an on-prem only installation page; we should have that be mapped to on-prem first, but also link out to Cloud and Kubernetes 
- Cluster: To install Determined, please follow … 
- CLI: Each user should also install the Determined command-line tools … 
14. Interact with the Cluster 
^ WebUI 
^ CLI 
Simplified version of the CLI page 
export DET_MASTER= 
det s 
link to the CLI page 
15. Training an example model 
^ Single GPU with example model 
^ Distributed training with example model 
^ Hyperparameter search with example model 
Exploring the Determined Web UI 
^ Experiment tracking 
^ Checkpointing 
^ Profiling 
^ Cluster management 
Prepare Data 
16. Add a separate Join the Community page 

## Test Plan




## Commentary (optional)




## Checklist

See [Release Note](../docs/release-notes/README.md) for details.





[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
